### DISCUSSION TITLE
Using dynamic links with Expo managed workflow

### DISCUSSION THREAD

**Discussion by josephphayden** [invertase/react-native-firebase#6685] [Q&A] [answered]
Hi, we're using the Expo managed workflow with a custom dev client but are using EAS build and are not managing any custom iOS or Android code (everything is managed in JS).

I've set up dynamic-links using the plugins as suggested and it works well on Android but not quite on iOS. Without access to the iOS code, I've not been able to complete all the setup steps listed [here](https://rnfirebase.io/dynamic-links/usage#ios-setup), specifically the update to `Associated Domains` and `info.plist` (for a custom domain).

I wanted to check if these steps are meant to be handled by the plugin, or whether we'll need to customise the iOS code ourselves to get it to work properly?

**Comment by mikehardy** [comment, marked as answer]
I may be wrong, but I was pretty sure Expo had either existing config plugins for those items or that they were simply supported in their app.json set of available customization keys already?
Certainly we've got Expo folks using Dynamic Links via EAS, so I know it is possible. There is even a config plugin for a currently needed workaround solving a race condition - for Expo people - a strong indicator people have this up and running in production / live apps --> #6650 + #6660

**Reply by josephphayden** [reply]
Thanks for the info. you're right that I can set the required fields in the `app.json`. My first time doing an Expo app  😄

**Reply by mikehardy** [reply]
Once you get it working (after the long configuration journey that is "setting up dynamic links"...) you might hit the edit button at top right of any page on rnfirebase.io and for the dynamic links module add an expo section (if one does not exist?) noting the keys you used - you can't be the only one curious about it, and Expo adoption here is relatively new (because config plugins are new...) so there are still lots of missing pieces of information. But I don't personally use Expo so I'm not great at helping unfortunately and need the community effort.

Good luck on your project!

**Reply by josephphayden** [reply]
Here you go: https://github.com/invertase/react-native-firebase/pull/6693. Hopefully might make it a bit clearer for people new to this Expo workflow, like me.
